lib/media_types/scheme.rb

```diff
@@ -385,24 +415,80 @@ def inspect(indentation = 0)
     end
 
     def assert_pass(fixture)
-      json = JSON.parse(fixture, { symbolize_names: true })
-
-      validate(json)
+      reduced_stack = caller_locations.filter { |location| !location.path.include?(__dir__) }
+      @fixtures << FixtureData.new(caller: reduced_stack.first, fixture: fixture, expect_to_pass: true)
     end
-    
+
     def assert_fail(fixture)
-      json = JSON.parse(fixture, { symbolize_names: true })
+      reduced_stack = caller_locations.filter { |location| !location.path.include?(__dir__) }
+      @fixtures << FixtureData.new(caller: reduced_stack.first, fixture: fixture, expect_to_pass: false)
+    end
+
+    def validate_scheme_fixtures(expect_symbol_keys, backtrace)
+      failed_fixtures = []
+      @fixtures.each do |fixture_data|
+        begin
+          validate_fixture(fixture_data, expect_symbol_keys, backtrace)
+        rescue UnexpectedValidationResultError => e
+          failed_fixtures << e.message
+        end
+      end
+      failed_fixtures
+    end
+
+    def validate_nested_scheme_fixtures(expect_symbol_keys, backtrace)
+      failed_fixtures = []
+      @rules.each do |key, rule|
+        next unless rule.is_a?(Scheme) || rule.is_a?(Links)
+
+        begin
+          rule.run_fixture_validations(expect_symbol_keys, backtrace.dup.append(key))
+        rescue AssertionError => e
+          failed_fixtures += e.fixture_errors
+        end
+      end
+      failed_fixtures
+    end
+
+    def validate_default_scheme_fixtures(expect_symbol_keys, backtrace)
+      return [] unless @rules.default.is_a?(Scheme)
+
+      @rules.default.run_fixture_validations(expect_symbol_keys, backtrace.dup.append('*'))
+      []
+    rescue AssertionError => e
+      return e.fixture_errors
+    end
+
+    def run_fixture_validations(expect_symbol_keys, backtrace = [])
+      failed_fixtures = []
+      failed_fixtures += validate_scheme_fixtures(expect_symbol_keys, backtrace)
+      failed_fixtures += validate_nested_scheme_fixtures(expect_symbol_keys, backtrace)
+      failed_fixtures += validate_default_scheme_fixtures(expect_symbol_keys, backtrace)
+
+      raise AssertionError.new(failed_fixtures) unless failed_fixtures.empty?
+
+      self.asserted_sane = true
+    end
+
+    def validate_fixture(fixture_data, expect_symbol_keys, backtrace = [])
+      json = JSON.parse(fixture_data.fixture, { symbolize_names: expect_symbol_keys })
+      expected_key_type = expect_symbol_keys ? Symbol : String
 
       begin
-        validate(json)
-      rescue MediaTypes::Scheme::ValidationError
-        return
+        validate(json, expected_key_type: expected_key_type, backtrace: backtrace)
+        unless fixture_data.expect_to_pass
```

Comment:
```suggestion
        unless fixture_data.expect_to_pass?
```

(Also update the `FixtureData` class)

Reply:
Fixed.